core/Tests/Tests.hs

```diff
@@ -189,6 +189,13 @@ prop_handshake_keyupdate = do
     params <- pick arbitraryPairParams
     runTLSPipeSimpleKeyUpdate params
 
+prop_handshake13_downgrade :: PropertyM IO ()
+prop_handshake13_downgrade = do
+    (cparam,sparam) <- pick arbitraryPairParams13
+    let debug' = (serverDebug sparam) { debugVersion = Just TLS12 }
+        sparam' = sparam { serverDebug = debug' }
+    runTLSInitFailure (cparam,sparam')
+
```

Comment:
I added the test case downgrading from TSL 1.3 to TLS 1.1.

If we can use `arbitraryPairParams` instead of `arbitraryPairParams13` and guess the negotiated version, we can cover downgrade senario (TLS 1.2 -> TLS 1.1). In this case, mixing no downgrade make sense to me.

Unfortunately, I have no idea on how to tell the negotiated version. So, this test covers downgrade senarios only.